frontend/packages/console-shared/src/components/Text/ClampedText.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import Measure from 'react-measure';
+import { Popover, Button } from '@patternfly/react-core';
+import './ClampedText.scss';
+
+type ClampedTextProps = {
+  content: string;
+  clampSize: number;
+};
+
+const ClampedText: React.FC<ClampedTextProps> = ({ content, clampSize = 1 }) => {
+  const [isContentClamped, setIsContentClamped] = React.useState<boolean>(false);
+  const style = { '--clamp-size': clampSize } as React.CSSProperties;
+
+  return (
+    <Measure
+      bounds
+      offset
+      scroll
+      onResize={(contentRect) => {
+        const {
+          scroll: { height: scrollHeight },
+          offset: { height: offsetHeight },
+        } = contentRect;
+        if (scrollHeight > offsetHeight) {
+          setIsContentClamped(true);
+        }
+      }}
+    >
+      {({ measureRef }) => (
+        <div ref={measureRef} className="ocs-clamped-text" style={style}>
+          {isContentClamped ? (
+            <Popover bodyContent={content}>
+              <Button variant="link">
+                <span className="ocs-clamped-text" style={style}>
+                  {content}
+                </span>
+              </Button>
```

Comment:
Using a button changes the text color as well as the alignment. We also don't need to duplicate the `ocs-clamped-text` class.
```suggestion
              <div tabIndex={0} role="button">
                {content}
              </div>
```

Reply:
done